src/components/OfficeSections/OfficeSectionsComponent.jsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Header,
+  Table,
+  Pagination,
+  Segment,
+  Divider,
+  Button
+} from 'semantic-ui-react';
+import _ from 'lodash';
+
+import TableRow from '../TableRowComponent';
+import NavBarComponent from '../../_components/NavBarContainer';
+import DropdownComponent from '../../_components/DropdownComponent';
+import LoaderComponent from '../../components/LoaderComponent';
+import rowOptions from '../../_utils/pageRowOptions';
+import ModalComponent from '../common/ModalComponent';
+import ItemsNotFoundComponent from '../common/ItemsNotFoundComponent';
+
+export class OfficeSectionsComponent extends React.Component {
+  state = {
+    activePage: 1,
+    limit: 10
+  };
+
+  componentDidMount() {
+    this.props.loadOfficeSections(this.state.activePage, this.state.limit);
+  }
+
+  handleRowChange = (e, data) => {
+    this.setState({ limit: data.value });
+    this.props.loadOfficeSections(this.state.activePage, data.value);
+  };
+
+  handlePaginationChange = (e, { activePage }) => {
+    this.setState({ activePage });
+    this.props.loadOfficeSections(activePage);
+  };
+
+  handleToggleModal = () => this.setState({ modalOpen: !this.state.modalOpen });
+
+  getTotalPages = () => Math.ceil(this.props.officeSectionsCount / this.state.limit);
+
+  render() {
+    if (this.props.isLoading) {
+      return (
+        <NavBarComponent>
+          <LoaderComponent />
+        </NavBarComponent>
+      );
+    }
+    if (!this.props.isLoading && _.isEmpty(this.props.officeSectionsList)) {
```

Comment:
`!this.props.isLoading` is redundant here. Because of line 46, we can be sure `isLoading = false` at this stage and only check the list